src/service/resources-impl.js

```diff
@@ -332,6 +332,27 @@ export class Resources {
   }
 
   /**
+   * Returns a promise to the layoutBox for the element. If the element is
+   * resource-backed then makes use of the resource layoutBox, otherwise
+   * measures the element directly.
+   * @param {!Element} element
+   * @return {!Promise<!../layout-rect.LayoutRectDef>}
+   */
+  getElementLayoutBox(element) {
```

Comment:
The point in this method is to allow an element to be measured regardless of whether it is a Resource instance associated with it or not, so it would probably be quite messy if this where in resource.js as it'd have to be a static method that then got a reference to Resources to make use of the vsync and viewport instances.